### PR TITLE
Update py-pennylane ecosystem to support v0.30.0

### PR DESCRIPTION
**Pull request by mlxd**
Adds support for [PennyLane v0.30.0](https://github.com/PennyLaneAI/pennylane/releases/tag/v0.30.0), [PennyLane Lightning v0.30.0](https://github.com/PennyLaneAI/pennylane-lightning/releases/tag/v0.30.0) and [Pennylane Lightning Kokkos v0.30.0](https://github.com/PennyLaneAI/pennylane-lightning-kokkos/releases/tag/v0.30.0).
Additionally, sets limits on the Kokkos versions to the 3.7.x release versions.
Fixes missing `py-pip` runtime dependency for PennyLane `qml.about()` functionality.

### CODE REVIEW

**Review by vincentmr — DISMISSED** [review]
Looks good to me.

File: var/spack/repos/builtin/packages/py-pennylane-lightning-kokkos/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pennylane-lightning-kokkos/package.py]: It says 0.28+ in the setup.py

File: var/spack/repos/builtin/packages/py-pennylane-lightning-kokkos/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pennylane-lightning-kokkos/package.py]: Seems like this should actually be a link dep?

File: var/spack/repos/builtin/packages/py-pennylane-lightning/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pennylane-lightning/package.py:55]: Should pybind11 be a link dep?

File: var/spack/repos/builtin/packages/py-pennylane-lightning-kokkos/package.py
Comment (vincentmr) [inline var/spack/repos/builtin/packages/py-pennylane-lightning-kokkos/package.py]: I think we don't want `kokkos<3.7`, no?

```suggestion
            f"kokkos@3.7+{backend.lower()}", when=f"+{backend.lower()}", type=("run", "build")
```

File: var/spack/repos/builtin/packages/py-pennylane-lightning-kokkos/package.py
Comment (mlxd) [reply var/spack/repos/builtin/packages/py-pennylane-lightning-kokkos/package.py]: Pybind11 is header-only, so any artifacts generated at build time should have everything needed to fully support import from Python. Should be fine to leave as "build" only here.

File: var/spack/repos/builtin/packages/py-pennylane-lightning/package.py
Comment (mlxd) [reply var/spack/repos/builtin/packages/py-pennylane-lightning/package.py:55]: As above, should be fine as "build" only.

File: var/spack/repos/builtin/packages/py-pennylane-lightning-kokkos/package.py
Comment (mlxd) [reply var/spack/repos/builtin/packages/py-pennylane-lightning-kokkos/package.py]: We have working with 3.6 in the past, but fine to fix this at 3.7 for now until 4.0 support is added. I'll update it.

File: var/spack/repos/builtin/packages/py-pennylane-lightning-kokkos/package.py
Comment (mlxd) [reply var/spack/repos/builtin/packages/py-pennylane-lightning-kokkos/package.py]: Good catch, yes this should be 0.28. I'll make an update.

File: var/spack/repos/builtin/packages/py-pennylane-lightning-kokkos/package.py
Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-pennylane-lightning-kokkos/package.py]: In Spack, "build" dep means add `<prefix>/bin` to the PATH, while "link" dep means inject `-L<prefix>/lib` and `-I<prefix>/include` via the compiler wrappers. So it sounds like link would make more sense here.

Comment (mlxd) [reply var/spack/repos/builtin/packages/py-pennylane-lightning-kokkos/package.py]: Ah, I misunderstood, thanks. I'll update this across the board. Thanks.

**Comment by adamjstewart** [comment]
Just a heads up, #37895 also affects many of these packages

**Review by adamjstewart — APPROVED** [review]
Looks fine to me. Let me know when maintainers have reached a general consensus and I can merge.

**Review by vincentmr — APPROVED** [review]
Looks good to me.

File: var/spack/repos/builtin/packages/py-pennylane-lightning-kokkos/package.py
Comment (vincentmr) [reply var/spack/repos/builtin/packages/py-pennylane-lightning-kokkos/package.py]: I see. I think we could set `kokkos@3.6:` then, instead of `kokkos@:3.7`. I was more worried about the lack of lower bound. In the next version however, we'll use `InitializationSettings` which appeared in Kokkos 3.7.

**Review by AmintorDusko — APPROVED** [review]
Looks good! Thank you, @mlxd for putting this together.
Thank you @vincentmr and @adamjstewart for your reviews.

**Comment by AmintorDusko** [comment]
> Just a heads up, #37895 also affects many of these packages

Hey @adamjstewart, thank you for the heads up. I will provide a review there once this PR is merged.

**Comment by mlxd** [comment]
Thanks @adamjstewart 
I reviewed #37895 no issue from me to proceed with the changes there.

**Pull request merged**
